resource_type_create.go

```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+)
+
+// ResourceTypeCreateParams represents the request to create a new resource type.
+type ResourceTypeCreateParams struct {
+	DisplayName     string          `json:"display_name"`
+	Description     string          `json:"description,omitempty"`
+	Scope           Scope           `json:"scope"`
+	RefreshStrategy RefreshStrategy `json:"refresh_strategy"`
+	SourceConfig    *SourceConfig   `json:"source_config,omitempty"`
+	IsEnabled       bool            `json:"is_enabled,omitempty"`
+}
+
+// CreateResourceType creates a new resource type.
+//
+// Note: requires a `Management` API key.
+func (c *Client) CreateResourceType(ctx context.Context, req *ResourceTypeCreateParams) (*ResourceType, error) {
+	rsp, err := c.makeRequest(ctx, http.MethodPost, "resource-types-v2", req)
```

Comment:
Can we drop the v2 from the path? (requires a backend change, but i don't see a conflict in the backend here either)